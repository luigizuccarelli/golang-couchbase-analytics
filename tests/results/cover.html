
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>handlers: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">lmzsoftware.com/lzuccarelli/golang-couchbase-analytics/pkg/handlers/handlers.go (100.0%)</option>
				
				<option value="file1">lmzsoftware.com/lzuccarelli/golang-couchbase-analytics/pkg/validator/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package handlers

import (
        "encoding/json"
        "fmt"
        "net/http"
        "os"
        "strconv"

        "lmzsoftware.com/lzuccarelli/golang-couchbase-analytics/pkg/connectors"
        "lmzsoftware.com/lzuccarelli/golang-couchbase-analytics/pkg/schema"
        //"github.com/gorilla/mux"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

func SankeyChartHandler(w http.ResponseWriter, r *http.Request, conn connectors.Clients) <span class="cov8" title="1">{
        var response schema.Response
        //vars := mux.Vars(r)

        addHeaders(w, r)

        // remove leading htpps:// and trailing journey_id for uniquness
        query := `SELECT        page.referrerName AS source,
                                                                                page.urlName AS destination,
                                                                                COUNT(message_id) AS count
                                                FROM PAGEEVENTS
                                                GROUP BY        page.referrerName as source,
                                                                                        page.urlName AS destination`

        ar, err := conn.AnalyticsQuery(query, nil)
        if err != nil </span><span class="cov8" title="1">{
                response = schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "ERROR", Message: fmt.Sprintf("Could not execute analytics query from couchbase %v", err)}
                w.WriteHeader(http.StatusInternalServerError)
                b, _ := json.MarshalIndent(response, "", "        ")
                fmt.Fprintf(w, string(b))
                return
        }</span>

        <span class="cov8" title="1">d, err := processSankeyResults(ar)
        if err != nil </span><span class="cov8" title="1">{
                response = schema.Response{Name: os.Getenv("NAME"), StatusCode: "500", Status: "ERROR", Message: fmt.Sprintf("Could not process analytics results from couchbase %v", err)}
                w.WriteHeader(http.StatusInternalServerError)
                b, _ := json.MarshalIndent(response, "", "        ")
                fmt.Fprintf(w, string(b))
                return
        }</span>

        <span class="cov8" title="1">conn.Debug(fmt.Sprintf("Analytics result from couchbase  %v \n", d))
        response = schema.Response{Name: os.Getenv("NAME"), StatusCode: "200", Status: "OK", Message: "Data retrieved succesfully", Sankey: d}
        w.WriteHeader(http.StatusOK)
        b, _ := json.MarshalIndent(response, "", "        ")
        fmt.Fprintf(w, string(b))</span>
}

func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        fmt.Fprintf(w, "{ \"version\" : \"1.0.2\" , \"name\": \""+os.Getenv("NAME")+"\" }")
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
}</span>

func processSankeyResults(res connectors.AnalyticsResult) ([]schema.SankeyData, error) <span class="cov8" title="1">{
        var row map[string]interface{}
        var data []schema.SankeyData

        for res.Next() </span><span class="cov8" title="1">{
                res.Row(&amp;row)
                if count, ok := row["count"]; ok </span><span class="cov8" title="1">{
                        x, _ := strconv.Atoi(fmt.Sprintf("%v", count))
                        d := schema.SankeyData{Value: x, From: fmt.Sprintf("%v", row["source"]), To: fmt.Sprintf("%v", row["destination"])}
                        data = append(data, d)
                }</span>
        }
        <span class="cov8" title="1">if err := res.Close(); err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>

        <span class="cov8" title="1">return data, nil</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package validator

import (
        "fmt"
        "github.com/microlib/simple"
        "os"
        "strconv"
        "strings"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *simple.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("Input paramaters -&gt; name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return fmt.Errorf(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span>

                <span class="cov8" title="1">logger.Error(fmt.Sprintf("%s envar is empty please set it", name))</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
func ValidateEnvars(logger *simple.Logger) error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "SERVER_PORT,true",
                "COUCHBASE_HOST,true",
                "COUCHBASE_DATABASE,true",
                "COUCHBASE_USER,true",
                "COUCHBASE_PASSWORD,true",
                "VERSION,true",
                "NAME,true",
                "COUCHBASE_BUCKET,true",
        }
        for x := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
