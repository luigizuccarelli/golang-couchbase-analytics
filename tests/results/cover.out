mode: set
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:12.59,16.27 4 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:24.2,24.12 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:16.27,17.15 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:22.3,22.69 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:17.15,20.4 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:29.50,41.23 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:46.2,46.12 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:41.23,42.54 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/validator/validate.go:42.54,44.4 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:20.90,30.16 6 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:38.2,39.16 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:47.2,51.27 5 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:30.16,36.3 5 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:39.16,45.3 5 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:54.54,56.2 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:59.57,65.2 4 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:67.88,71.17 3 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:79.2,79.36 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:83.2,83.18 1 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:71.17,73.36 2 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:73.36,77.4 3 1
gitea-cicd.apps.aws2-dev.ocp.14west.io/cicd/trackmate-couchbase-analytics/pkg/handlers/handlers.go:79.36,81.3 1 1
